### PR TITLE
add e6data query runner with test cases

### PR DESCRIPTION
**Pull request by anushae6**
## What type of PR is this? 

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [X] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## How is this tested?

- [X] Unit tests (pytest, jest)
- [X] E2E Tests (Cypress)
- [X] Manually
- [ ] N/A

### CODE REVIEW

**Comment by masayuki038** [comment]
@anushae6 Sorry for the delay. 
Probably none of us have ever touched e6data. So all we rely on is your test results. I'm concerned that "Manually" in "How is this tested?" is not checked. Have you connected to e6data in your branch and checked the results of the query?

**Comment by masayuki038** [comment]
@justinclift @guidopetri We are discussing [python module dependencies](https://github.com/getredash/redash/discussions/6591). As a result of my checks, I don't think there is any problem in that point.　Are there any other points to check about accepting new data sources?

**Comment by anushae6** [comment]
My bad,I forgot to check that box. I have tested it manually as well. I'll attach some screenshots for the same
e6data_datasource_addition
e6data_configuration
query_result

**Comment by masayuki038** [comment]
@anushae6 Thanks! I confirmed your screenshots. Looks good.
Please give us a little time to discuss this. thank you.

**Comment by anushae6** [comment]
> @anushae6 Thanks! I confirmed your screenshots. Looks good. Please give us a little time to discuss this. thank you.

Sure, thanks!

**Comment by masayuki038** [comment]
@anushae6 I checked your updates. 

(Optional) It's even better to disassemble run_query and test parts of it.

When registering a new query_runner, it is desirable to have more test coverage. The next person who maintains this code will be helpful 😄 Of course, I think this is optional.

Please consider it if you have time. thank you.

**Review by masayuki038 — APPROVED** [review]

**Comment by anushae6** [comment]
> @anushae6 I checked your updates.
> 
> (Optional) It's even better to disassemble run_query and test parts of it.
> 
> When registering a new query_runner, it is desirable to have more test coverage. The next person who maintains this code will be helpful 😄 Of course, I think this is optional.
> 
> Please consider it if you have time. thank you.

Sure, I didn't add test case for run_query because I'll be patching most of the code so thought it won't be that useful. Will work on all the feedbacks and modify the PR.

File: pyproject.toml [hunk 1]
```diff
@@ -173,4 +174,4 @@ select = ["C9", "E", "F", "W", "I001"]
 max-complexity = 15
 
 [tool.ruff.per-file-ignores]
-"__init__.py" = ["F401"]
+"__init__.py" = ["F401"]
```
Comment (guidopetri) [inline pyproject.toml:177]: We'll want to roll back this small change :)

File: redash/query_runner/e6data.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+logger = logging.getLogger(__name__)
+
+E6DATA_TYPES_MAPPING = {
+    "INT": TYPE_INTEGER,
+    "BYTE": TYPE_INTEGER,
+    "INTEGER": TYPE_INTEGER,
+    "LONG": TYPE_INTEGER,
+    "SHORT": TYPE_INTEGER,
+    "FLOAT": TYPE_INTEGER,
+    "DOUBLE": TYPE_INTEGER,
```
Comment (guidopetri) [inline redash/query_runner/e6data.py:36]: These don't seem right to me - should this be `TYPE_FLOAT`?

**Review by guidopetri — CHANGES_REQUESTED** [review]
This looks reasonable to me in general. However, we definitely want to add more tests for this query runner - ideally we have at least a 90% patch coverage.

**Comment by anushae6** [comment]
Added all the requested changes, please review.

**Comment by masayuki038** [comment]
@anushae6 Thanks for adding a test for `run_query`!

**Comment by anushae6** [comment]
I'm not sure why the filter spec test is failing in e2e tests. When I ran the e2e tests on local, it passed all tests

**Review by guidopetri — APPROVED** [review]
@anushae6 thanks so much for your work on this PR!

File: tests/query_runner/test_e6data.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+
+@patch("e6data_python_connector.e6data_grpc.Cursor")
+def test_test_connection(mock_cursor):
+    query = "SELECT 1"
+    user = None
+    mock_cursor.return_value.fetchall.return_value = [[1]]
+    mock_cursor.return_value.description = [("EXPR$0", "INTEGER", None, None, None, None, True)]
+
+    json_data, error = runner.run_query(query, user)
```
[the remark this answers is not in the document]
Comment (masayuki038) [reply tests/query_runner/test_e6data.py:51]: @anushae6 It has been merged, so please do this at another time. thank you!

**Pull request merged**